cmd/intervalSubsetMatrix/intervalSubsetMatrix.go

```diff
@@ -32,17 +33,45 @@ func intervalSubsetMatrix(unionFile string, fileListFile string, outFile string)
 	}
 	unionTree := interval.BuildTree(unionIntervals)
 
-	var mat = make(map[string][]int) //this will be our output feature matrix. mat[row] => col, where rows are genomic regions, cols are files.
+	//this will be our output feature matrix. mat[row] => col, where rows are genomic regions, cols are files.
+	var mat = make(map[string][]float64)
+
+	// if in markMultipleOverlaps mode, create output feature matrix for additional multiple overlap output file
+	var matMultipleOverlaps = make(map[string][]float64)
+
 	for _, u := range unionIntervals {
-		mat[interval.CoordsToString(u)] = make([]int, len(files)) //initialize all rows to zeros based on the number of files. Map is indexed by coords of regions.
+		mat[interval.CoordsToString(u)] = make([]float64, len(files)) //initialize all rows to zeros based on the number of files. Map is indexed by coords of regions.
+
+		if markMultipleOverlaps != "" {
+			matMultipleOverlaps[interval.CoordsToString(u)] = make([]float64, len(files))
+		}
+
 	}
 
 	for i = range files {
 		currIntervalChan = interval.GoReadToChan(files[i])
 		for j = range currIntervalChan {
-			currOverlaps = interval.Query(unionTree, j, "any")
+			currOverlaps = interval.Query(unionTree, j, "any") // each j (1 bed region from the file of the column) can overlap 0 or 1 or more bed region of the row
 			if len(currOverlaps) > 0 {
-				mat[interval.CoordsToString(currOverlaps[0])][i] = 1
+				for _, k := range currOverlaps { // k >= 1 would mean j overlaps more than 1 bed region of the row
+					if fraction {
+						overlapSize = interval.OverlapSize(k, j)
+						mat[interval.CoordsToString(k)][i] += float64(overlapSize) / float64(interval.IntervalSize(k)) // add number rather than assign number, in case 1 bed region of the row overlaps multiple bed regions in the files of the columns
```

Comment:
So if I understand correctly, we are reporting the proportion of bases in a haqer element that overlap an interval set. Is that correct? When haqer window sizes are very large even if the haqer overlaps an small interval completely, it will return a small proportion. Is that the functionality you want?

Reply:
Yes!

Reply:
great! Everything looks good on my end. approving now